kernel/kernel-api/src/test/scala/io/delta/kernel/internal/clustering/ClusteringMetadataDomainSuite.scala

```diff
@@ -0,0 +1,140 @@
+
+    assert(clusteringMD.getClusteringColumnsAsColumnList === List(new Column(Array(
+      "col1",
+      "`col2,col3`",
+      "`col4.col5`,col6"))).asJava)
+  }
+
+  test("Successfully get DomainMetadata for non-nested columns") {
+    val schema = new StructType()
+      .add("id", IntegerType.INTEGER, true)
+      .add("name", IntegerType.INTEGER, true)
+      .add("age", IntegerType.INTEGER, true)
+
+    val clusterColumns = List(new Column("name"), new Column("age"))
+    val physicalColumns = convertToPhysicalColumn(clusterColumns, schema)
+
+    val clusteringMetadataDomain =
+      ClusteringMetadataDomain.fromPhysicalColumns(
+        physicalColumns.asJava)
+
+    val clusteringDomainMetadata = clusteringMetadataDomain.toDomainMetadata
+    assert(clusteringMetadataDomain.getClusteringColumnsAsColumnList == clusterColumns.asJava)
+    assert(clusteringDomainMetadata.getDomain == "delta.clustering")
+    assert(clusteringDomainMetadata.getConfiguration ==
+      """{"clusteringColumns":[["name"],["age"]]}""")
+  }
+
+  test("Successfully get DomainMetadata for nested columns") {
+    val schema = new StructType()
+      .add("id", IntegerType.INTEGER, true)
+      .add(
+        "user",
+        new StructType()
+          .add(
+            "address",
+            new StructType()
+              .add("city", StringType.STRING, true)))
+
+    val clusterColumns = List(new Column(Array("user", "address", "city")))
+    val physicalColumns = convertToPhysicalColumn(clusterColumns, schema)
+
+    val clusteringMetadataDomain = ClusteringMetadataDomain.fromPhysicalColumns(
+      physicalColumns.asJava)
+
+    val clusteringDomainMetadata = clusteringMetadataDomain.toDomainMetadata
+    assert(clusteringMetadataDomain.getClusteringColumnsAsColumnList ==
+      List(new Column(Array("user", "address", "city"))).asJava)
+    assert(clusteringDomainMetadata.getDomain == "delta.clustering")
+    assert(clusteringDomainMetadata.getConfiguration ==
+      """{"clusteringColumns":[["user","address","city"]]}""")
+  }
+
+  test("Correctly maps logical column names to physical column names") {
+    val schema = new StructType()
+      .add("id", IntegerType.INTEGER, true)
+      .add("name", IntegerType.INTEGER, true)
+
+    val metadata =
+      ColumnMapping.updateColumnMappingMetadataIfNeeded(
+        createMetadata(schema).withColumnMappingEnabled("id"),
+        true)
+
+    val clusterColumns = List(new Column("name"))
+    val physicalColumns = convertToPhysicalColumn(clusterColumns, metadata.get.getSchema)
+
+    val clusteringMetadataDomain = ClusteringMetadataDomain.fromPhysicalColumns(
+      physicalColumns.asJava)
+
+    assert(clusteringMetadataDomain.toDomainMetadata.getDomain == "delta.clustering")
+    assert(clusteringMetadataDomain.getClusteringColumnsAsColumnList.asScala.size == 1)
+    assert(clusteringMetadataDomain.getClusteringColumnsAsColumnList.get(
+      0).getNames()(0).startsWith("col-"))
```

Comment:
given you hvat the converted metadata, should be able to get the compare the actual physical col name.